common/test.c

```diff
@@ -501,8 +503,11 @@ p11_test_copy_setgid (const char *input)
 		return NULL;
 	}
 
-	path = strdup ("/tmp/test-setgid.XXXXXX");
-	assert (path != NULL);
+	tmpdir = getenv("TMPDIR");
+	if (tmpdir == NULL)
+		tmpdir = "/tmp";
+
+	assert(asprintf(&path, "%s/test-setgid.XXXXXX", tmpdir) >= 0);
```

Comment:
It would be good to avoid side-effect in ```assert```, since the code can be omitted at compile time, when NDEBUG is set. Also please put a space after function name for style consistency.